src/go/configmanager/config_manager.go

```diff
@@ -138,7 +137,13 @@ func NewConfigManager(mf *metadata.MetadataFetcher, opts options.ConfigGenerator
 
 	}
 
-	if m.serviceConfigFetcher, err = sc.NewServiceConfigFetcher(mf, opts, m.serviceName); err != nil {
+	var accessTokenFromImds sc.GetAccessTokenFunc
+	if mf != nil {
+		accessTokenFromImds = func() (string, time.Duration, error) { return mf.FetchAccessToken() }
+	}
```

Comment:
What happens if it's nil? Wouldn't this cause a panic in line 146 when you pass a function without a behavior?